### ISSUE TITLE
Cannot Properly Resume Training from Loaded Checkpoint

### ISSUE THREAD

**Issue by SaviNimz** [pyfenn/fenn#52]
The current Trainer implementation does not correctly support resuming training from a loaded checkpoint. Although load_checkpoint() returns the saved epoch number, the fit() method always starts training from:

`for epoch in range(self._epochs):`

**Comment by spencerirving** [comment]
Hey @SaviNimz! 

I worked on this implementation of the checkpointing and see your issue.

If @blkdmr is ok with it I can take care of this issue!

**Comment by blkdmr** [comment]
@spencerirving has this issue been fixed?

**Comment by blkdmr** [comment]
@spencerirving I proposed a small update to the Trainer class a few days ago on the Discord server, but you might not have seen it. I suggested adding the ability to pass a validation set to the train function. When this feature is added, you might consider switching the “best” criterion from lowest loss to highest validation accuracy, since a lower loss can sometimes be due to overfitting. If you’d like to stay updated, feel free to join the Discord server.
